web/modules/custom/server_general/src/ElementTrait.php

```diff
@@ -69,15 +69,17 @@ protected function buildElementCta(string $title, array $body, string $button_te
    *   The title.
    * @param string $subtitle
    *   The subtitle.
-   * @param string $button_text
+   * @param string|null $button_text
    *   The button text.
-   * @param \Drupal\Core\Url $url
+   *   If NULL, no button is rendered. Defaults to NULL.
+   * @param \Drupal\Core\Url|null $url
    *   The URL to link the button to.
+   *   If NULL, no button is rendered. Defaults to NULL.
    *
    * @return array
    *   Render array.
    */
-  protected function buildElementHeroImage(array $image, string $title, string $subtitle, string $button_text, Url $url): array {
+  protected function buildElementHeroImage(array $image, string $title, string $subtitle, string $button_text = NULL, Url $url = NULL): array {
```

Comment:
How about changing this to `Link $link`. This means we won't have weird cases where we have button text, but no button URL.

```suggestion
  protected function buildElementHeroImage(array $image, string $title, string $subtitle, Link $link): array {
```

Reply:
Ok, I have amended the commit, please take another look. Notice the `Link` object must have a default `NULL` value.

Also updated the `StyleGuideController` to reflect the change in the parameters passed on to `ElementTrait::buildElementHeroImage`.